packages/sign-client/src/controllers/engine.ts

```diff
@@ -572,6 +592,408 @@ export class Engine extends IEngine {
+        iat: new Date().toISOString(),
+        exp,
+        nbf,
+        resources,
+      },
+      requester: { publicKey, metadata: this.client.metadata },
+      expiryTimestamp,
+    };
+
+    // ----- build namespaces for fallback session proposal ----- //
+    const namespaces = {
+      eip155: {
+        chains,
+        // request `personal_sign` method by default to allow for fallback siwe
+        methods: [...new Set(["personal_sign", ...methods])],
+        events: ["chainChanged", "accountsChanged"],
+      },
+    };
+
+    const proposal = {
+      requiredNamespaces: {},
+      optionalNamespaces: namespaces,
+      relays: [{ protocol: "irn" }],
+      proposer: {
+        publicKey,
+        metadata: this.client.metadata,
+      },
+      expiryTimestamp,
+    };
+
+    const { done, resolve, reject } = createDelayedPromise(
+      ENGINE_RPC_OPTS.wc_sessionAuthenticate.req.ttl,
+      "Request expired",
+    );
+
+    // handle fallback session proposal response
+    const onSessionConnect = async ({ error, session }: any) => {
+      // cleanup listener for authenticate response
+      this.events.off(engineEvent("session_request", id), onAuthenticate);
+      if (error) reject(error);
+      else if (session) {
+        session.self.publicKey = publicKey;
+        await this.client.session.set(session.topic, session);
+        await this.setExpiry(session.topic, session.expiry);
+        if (pairingTopic) {
+          await this.client.core.pairing.updateMetadata({
+            topic: pairingTopic,
+            metadata: session.peer.metadata,
+          });
+        }
+        const sessionObject = this.client.session.get(session.topic);
+        resolve({
+          session: sessionObject,
+        });
+      }
+    };
+    // handle session authenticate response
+    const onAuthenticate = async (payload: any) => {
+      // remove cleanup for fallback response
+      this.events.off(engineEvent("session_connect"), onSessionConnect);
+      if (payload.error) {
+        // ignore unsupported method error
+        const error = getSdkError("WC_METHOD_UNSUPPORTED", "wc_sessionAuthenticate");
+        if (payload.error.code === error.code) return;
+        return reject(payload.error.message);
+      }
+      const {
+        cacaos,
+        responder,
+      }: {
+        cacaos: AuthTypes.SessionAuthenticateResponseParams["cacaos"];
+        responder: AuthTypes.SessionAuthenticateResponseParams["responder"];
+      } = payload.result;
+
+      const pendingRequest = this.client.auth.requests.get(id);
+      if (!pendingRequest) {
+        throw new Error(`Could not find pending auth request with id ${id}`);
+      }
+
+      const approvedMethods: string[] = [];
+      const approvedAccounts: string[] = [];
+      for (const cacao of cacaos) {
+        const isValid = await validateSignedCacao({ cacao, projectId: this.client.core.projectId });
+        if (!isValid) {
+          this.client.logger.error(cacao);
```

Comment:
Do we need more context on the error rather than forwarding just the failed cacao?